Src/FluentAssertions/Equivalency/AssertionRuleEquivalencyStep.cs

```diff
@@ -8,16 +8,18 @@ namespace FluentAssertions.Equivalency
 {
     public class AssertionRuleEquivalencyStep<TSubject> : IEquivalencyStep
     {
-        private readonly Expression<Func<IMemberInfo, bool>> predicate;
+        private readonly Func<IMemberInfo, bool> predicate;
+        private readonly string description;
         private readonly Action<IAssertionContext<TSubject>> assertion;
         private readonly AutoConversionStep converter = new AutoConversionStep();
 
         public AssertionRuleEquivalencyStep(
             Expression<Func<IMemberInfo, bool>> predicate,
             Action<IAssertionContext<TSubject>> assertion)
         {
-            this.predicate = predicate;
+            this.predicate = predicate.Compile();
```

Comment:
I think it is confusing to use the same variable/member name for different types in one method.

Reply:
I don't disagree, but at least it's now consistent with:
* PredicateBasedOrderingRule.cs
* ExcludeMemberByPredicateSelectionRule.cs
* IncludeMemberByPredicateSelectionRule.cs